dags/synapse_minimal_jsonld_dag.py

```diff
@@ -0,0 +1,399 @@
+                return
+
+            df = DataFrame(
+                data={
+                    "dataset": [dataset_id],
+                    "minimal_croissant_file_s3_object": [s3_url]
+                }
+            )
+            if existing_row_df.empty:
+                # If the row does not exist, create a new row
+                print("Creating new row in Synapse table")
+                Table(id=SYNAPSE_TABLE_FOR_CROISSANT_LINKS).store_rows(values=df)
+                
+            else:
+                # Update the existing row with the new value
+                print("Updating existing row in Synapse table")
+                existing_row_df["minimal_croissant_file_s3_object"] = [s3_url]
+                Table(id=SYNAPSE_TABLE_FOR_CROISSANT_LINKS).store_rows(values=existing_row_df)
+
+        except Exception as ex:
+            otel_logger.exception(
+                "Failed to push croissant file link to Synapse.")
+            otel_tracer.span_processor.force_flush()
+            otel_logger.handlers[0].flush()
+            raise ex
+
+    @task
+    def create_and_save_jsonld(**context) -> None:
+        """
+        Create and save the minimal croissant JSON-LD files to S3 and links to a synapse table for all datasets in the home page of data catalog.
+
+        Arguments:
+            dataset_collection: The dataset collection to query for datasets.
+
+        Returns:
+            The list of dataset IDs for the given dataset collection.
+        """
+        syn_hook = SynapseHook(context["params"]["synapse_conn_id"])
+        syn_client: Synapse = syn_hook.client
+
+        table = query(f"select * from {SYNAPSE_DATA_CATALOG}", synapse_client=syn_client)
+        
+        push_to_s3 = context["params"]["push_results_to_s3"]
+        push_to_synapse = context["params"]["push_links_to_synapse"]
+
+        
+        for index, row in table.iterrows():
+            if pd.isnull(row["id"]):
+                # skip rows without a Synapse ID
+                continue
+
+            dataset_id = row["id"]
+            dataset_name = row["name"]
+            dataset = syn_client.get(dataset_id, downloadFile=False)
+
+            link = f"https://www.synapse.org/#!Synapse:{row['id']}"
+
+            minimal_croissant_file = {
+                "@context": "https://schema.org/",
+                "@type": "Dataset",
+                "name": dataset_name,
+                "description": "" if pd.isnull(row["description"]) else row["description"],
+                "url": link,
+                "identifier": dataset_id,
+                "creator": {
+                    "@type": "Organization",
+                    "name": "Sage Bionetworks" if pd.isnull(row["contributors"]) else row["contributors"]
+                },
+                "license": "" if pd.isnull(row["license"]) else row["license"]
+            }
+
+            s3_key = f"{dataset_name}_{dataset_id}.minimal_croissant.jsonld"
```

Comment:
The version has to be encoded into the name of this file name. This is because all the files are uploaded into the same location. It is perfectly valid for multiple versions of the same Dataset to exist in the same DatasetCollection.

`s3_key = f"{dataset.name}_{dataset_id}.{dataset_version}_datasetCollection_{dataset_collection}_croissant.jsonld"` is the format I used in the other script.

Reply:
@linglp version tracking is not required at this time since some of these data catalog datasets point to synapse folders